Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnSlide.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class OnSlide : OnMovementSpecial
+{
+
+    [SerializeField]
+    private float wallSlideUpForce;
+    [SerializeField]
+    private float wallSlideDownForce;
+    [SerializeField]
+    private float wallSlideUpDecay;
+    [SerializeField]
+    private float wallSlideDownDecay;
+    [SerializeField]
+    private Vector2 groundSlideForce;
+    [SerializeField]
+    private float groundSlideDecay;
+
+    [SerializeField]
+    private PlayRandomSource wallSlideSound;
+
+    private bool goingRight;
+
+    private bool goingUp;
+
+    private Vector3 appliedForce;
+
+    private Vector2 test;
+
+    private bool isOnWall;
+    protected override void Start()
+    {
+        base.Start();
+        goingUp = false;
+        goingRight = false;
+    }
+
+    public override Vector3 ApplyFriction(Vector3 velocity)
+    {
+        if (isDisabled || !isInUse )
+        {
+            return GetSimulatedState().ApplyFriction(velocity);
+        }
+
+        test = player.AimDirection;
+
+        goingRight = velocity.x > 0;
+        goingUp = velocity.y > 0;
+        if (velocity.x == 0) goingRight = player.AimDirection.x > 0;
+
+        if (GetSimulatedState() == GetComponent<OnWall>())
+        {
+            velocity.x = 0;
+            goingRight = controller.collisions.Left;
+            //decrease in speed going up, increase going down
```

Comment:
I envision the decay as small, but I want it to be there if we feel slide is too good. I'll also probably eventually add in the ability to jump out of slides in some capacity later on once we know how it feels.